packages/build/bin/build.js

```diff
@@ -1,13 +1,40 @@
 #!/usr/bin/env node
 
-const {
-  build
-} = require('../dist/src/build')
-const logSuccess = () => console.log('done')
-const logError = (error) => console.log('error', error) && process.exit(1)
+const helpText = `
+  --mode edge|connected         # Sets MODE
+  --stage test|prod             # ... not supported right now
+  --compatible false            # Sets Babel transform
+  --optimize-react true|false   # Sets NODE_ENV to production
+  --optimize-bundle false       # Minimize, dead code elimination etc.
+  --type-check false            # Transpile only?
+  -h                            # show this
+`
+
+if (process.argv.join().includes('-h')) {
+  console.info(helpText)
+  process.exit(0)
+}
+
+function getArgv(setting, def) {
+  const index = process.argv.indexOf(setting) + 1
+  const v = index !== 0 ? process.argv[index] : def
+  return v === 'true' || v === 'false' ? v === 'true' : v
+}
 
-const appPath = process.cwd()
 const opts = {
-  vendor: ['react', 'react-dom', 'react-helmet', 'lodash', 'typestyle'],
+  mode: getArgv('--mode', process.env.MODE),
+  stage: getArgv('--stage', process.env.STAGE),
+  compatible: getArgv('--compatible', false),
+  optimizeReact: getArgv('--optimize-react', false),
+  optimizeBundle: getArgv('--optimize-bundle', false),
+  typeCheck: getArgv('--type-check', true),
 }
-build(appPath, opts).then(() => logSuccess()).catch(error => logError(error))
\ No newline at end of file
+
+opts.mode ? (process.env.MODE = opts.mode) : null
+process.env.STAGE = opts.stage
```

Comment:
would work but afaik it wont be used currently because no one uses STAGE (or whatever at least pleasepleaseplease dont use NODE_ENV for that....)

a problem we need to fix elsewhere

Reply:
we should definitely discuss this in the future